ui/app/templates/components/dashboard/secrets-engines-card.hbs

```diff
@@ -1,80 +1,100 @@
-<Hds::Card::Container @hasBorder={{true}} class="has-padding-l secrets-engines-card" data-test-card="secrets-engines">
+<Hds::Card::Container
+  @hasBorder={{true}}
+  class="has-padding-l {{if this.filteredSecretsEngines.length 'secrets-engines-card'}}"
+  data-test-card="secrets-engines"
+>
   <div class="is-flex-between">
     <h3 class="title is-4 has-left-margin-xxs" data-test-dashboard-secrets-engines-header>Secrets engines</h3>
 
-    <LinkTo class="is-no-underline has-right-margin-xxs" @route="vault.cluster.secrets.backends">
-      Details
-    </LinkTo>
+    {{#if this.filteredSecretsEngines.length}}
```

Comment:
same comment here about these conditionals - this could just be `this.filteredSecretsEngines`